sql/select.go

```diff
@@ -251,16 +249,18 @@ func (v *indexInfo) init(s *scanNode) {
 		v.cost = float64(1 + len(v.desc.Columns) - len(v.desc.PrimaryIndex.ColumnIDs))
 	} else {
 		v.cost = 1
+		if !v.covering {
+			v.cost += float64(1 + len(v.desc.Columns) - len(v.desc.PrimaryIndex.ColumnIDs))
```

Comment:
How can it be negative? The primary index cannot have more columns than are in the table.